lib/tjbot.js

```diff
@@ -1137,52 +1162,73 @@ TJBot.prototype.speak = function(message) {
+    }
+
 
-    return new Promise(function(resolve, reject) {
-        // if we don't have a speaker, throw an error
-        if (self._soundplayer == undefined) {
-            reject(new Error("unable to play audio, TJBot hardware doesn't include a \"speaker\""));
-            return;
-        }
 
-        // initialize soundplayer lib
-        var speakerOptions = {
-            filename: soundFile,
-            gain: 100,
-            debug: true,
-            player: "aplay", // "afplay" "aplay" "mpg123" "mpg321"
-            device: self.configuration.speak.speakerDeviceId
+    if (self.isPlaying) {
+
+        if (self.configuration.speak.queueSpeech || self.playerOptions.queueSpeech) {
+            winston.debug(this.speechQueue.length + " items in Queue. Queing audio file with parameter ", self.playerOptions);
+            winston.verbose("You attempted to play audio while ", self.playerOptions.filename, " file was playing. Queing ");
+            this.speechQueue.push(self.playerOptions)
+        } else {
+            winston.verbose("You attempted to play audio file ", self.playerOptions.filename, " while a file was playing. To enable Queuing, add `speak: {queueSpeech: true}` to tjconfig ");
         }
-        var player = new self._soundplayer(speakerOptions);
+    } else {
+        self.isPlaying = true;
+        // pause listening while we play a sound -- using the internal
+        // method to avoid a capability check (and potential fail if the TJBot
+        // isn't configured to listen)
+        self._pauseListening();
 
-        winston.debug("Playing audio with parameters: ", speakerOptions);
+        return new Promise(function(resolve, reject) {
+            // if we don't have a speaker, throw an error
+            if (self._soundplayer == undefined) {
+                reject(new Error("unable to play audio, TJBot hardware doesn't include a \"speaker\""));
+                return;
+            }
 
-        player.on('complete', function() {
-            winston.debug("audio playback finished");
+            var player = new self._soundplayer(self.playerOptions);
 
-            // resume listening
-            self._resumeListening();
+            winston.debug("Playing audio with parameters: ", self.playerOptions);
 
-            // done
-            resolve();
-        });
+            player.on('complete', function() {
+                winston.debug("audio playback finished");
 
-        player.on('error', function(err) {
-            winston.error('Error occurred while playing audio :', err);
-        });
+                // resume listening
+                self._resumeListening();
+                self.isPlaying = false
 
-        // play the audio
-        player.play(soundFile);
-    });
+                // done
+                resolve();
+
+                //process elements in speech queue
+                if (self.speechQueue.length > 0) {
```

Comment:
what happens if there are multiple things queued?  should this be a `while` loop?